src/ts/deploy.ts

```diff
@@ -222,31 +222,55 @@ export async function prepareSafeDeployment(
   realTokenAddress: string;
   virtualTokenAddress: string;
 }> {
-  const { safeTransaction: realTokenDeployment, address: realTokenAddress } =
-    await getDeploymentTransaction(
-      ContractName.RealToken,
-      realTokenDeployParams,
+  const {
+    safeTransaction: realTokenDeployTransaction,
+    address: realTokenAddress,
+  } = await getDeploymentTransaction(
+    ContractName.RealToken,
+    realTokenDeployParams,
+    ethers,
+    salt,
+  );
+
+  const { virtualTokenDeployTransaction, virtualTokenAddress } =
+    await prepareVirtualSafeDeployment(
+      { ...virtualTokenDeployParams, realToken: realTokenAddress },
       ethers,
       salt,
     );
+
+  return {
+    realTokenDeployTransaction,
+    virtualTokenDeployTransaction,
+    deployTransaction: multisend(
+      [realTokenDeployTransaction, virtualTokenDeployTransaction],
+      multisendAddress,
+    ),
+    realTokenAddress,
+    virtualTokenAddress,
+  };
+}
+
+export async function prepareVirtualSafeDeployment(
```

Comment:
THe first time I was reading this function name, I thought you are preapring  safe contract deployment. Maybe `prepareVirtaulDeploymentFromSafe()`  would be better.